ebpf_ffi/cbpf.cc

```diff
@@ -75,22 +81,24 @@ struct bpf_result ffi_load_cbpf_program(void *prog_buff, int coverage_enabled,
 
   if (socks[0] < 0) {
     // Return why we failed to load the program.
-    vres.set_bpf_error(error_message);
-    vres.set_is_valid(false);
-  } else {
-    vres.set_is_valid(true);
+    return validation_error(error_message, &vres);
   }
+  vres.set_is_valid(true);
 
   return serialize_proto(vres);
 }
 
 bool execute_cbpf_program(int socket_parent, int socket_child, uint8_t *input,
-                          int input_length, std::string *error_message) {
-  if (write(socket_child, input, input_length) != input_length) {
+                          uint8_t *output, int input_length,
+                          std::string *error_message) {
+  if (write(socket_parent, input, input_length) != input_length) {
     *error_message = "Could not write all data to socket";
     return false;
   }
 
+  if (read(socket_child, output, input_length) != input_length) {
```

Comment:
could it be the case where the socket filter drops the message we send on write and then this function keeps waiting for a message forever?

Could we try closing sock_parent before this one and then testing manually the case where a filter drops the packet and when the packet is not dropped?

Reply:
Done, timeout for `setsocketopt` was added